main.go

```diff
@@ -14,50 +14,21 @@ import (
 	"github.com/antonlindstrom/mesos_stats"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/log"
+	"github.com/pvdissel/mesos_metrics"
 )
 
 const concurrentFetch = 100
 
 // Commandline flags.
 var (
-	addr           = flag.String("web.listen-address", ":9105", "Address to listen on for web interface and telemetry")
-	autoDiscover   = flag.Bool("exporter.discovery", false, "Discover all Mesos slaves")
-	localURL       = flag.String("exporter.local-url", "http://127.0.0.1:5051", "URL to the local Mesos slave")
-	masterURL      = flag.String("exporter.discovery.master-url", "http://mesos-master.example.com:5050", "Mesos master URL")
-	metricsPath    = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics")
+	addr = flag.String("web.listen-address", ":9105", "Address to listen on for web interface and telemetry")
+	autoDiscover = flag.Bool("exporter.discovery", false, "Discover all Mesos slaves")
+	localURL = flag.String("exporter.local-url", "http://127.0.0.1:5051", "URL to the local Mesos slave")
+	masterURL = flag.String("exporter.discovery.master-url", "http://mesos-master.example.com:5050", "Mesos master URL")
```

Comment:
No, you'll get the data from the complete cluster based on the elected master of that cluster. But all masters should be in-sync with the elected master. Or... is that exactly what you want to know.. hmm.. good point.
In the original code it already went to the elected master and started from there, but maybe good to not go to the elected master but start at that specific master that's provided.
Not sure how that would work in the metrics gathering, as you're actually fetching the same info from different places. And there may-be a difference in values because of different times of scraping and possibly not completely in-sync masters.

What do you suggest?